src/netaddress.h

```diff
@@ -161,7 +161,7 @@ class CService : public CNetAddr
         inline void SerializationOp(Stream& s, Operation ser_action) {
             READWRITE(ip);
             unsigned short portN = htons(port);
-            READWRITE(FLATDATA(portN));
+            READWRITE(portN);
```

Comment:
Mhm.. I don't think we can simply do this.
This is not a char (or char-array), for which we introduced the native serialization. This is a short int, stored in memory in native byte order, but serialized in big endian notation (see bitcoin#12886 and bitcoin#12916)

Reply:
hmm yeah, nice catch.